lib/src/git_backend.rs

```diff
@@ -537,10 +538,16 @@ fn commit_from_git_without_root_parent(
             .map(|b| b.reverse_bits())
             .collect(),
     );
-    let parents = commit
-        .parents()
-        .map(|oid| CommitId::from_bytes(oid.as_bytes()))
-        .collect_vec();
+    // shallow commits don't have parents their parents actually fetched, so we
+    // discard them here
+    let parents = if is_shallow {
+        vec![]
+    } else {
```

Comment:
Maybe this means that the shallow-boundary commit was rewritten, its parents would be permanently lost?

That might be okay for experimental support of shallow clones, but we'll need to find a better workaround later.

Reply:
As in, if someone tries to unshallow their repository we won't pick up those changes? I'll have to check but I imagine we can work around that by marking shallow commits or so.

Either way I'd also say for initial support that would be be probably fine.

Reply:
Yes, not crashing is at least better than crashing.

If the repo is later unshallowed (or just deepened a bit), we would somehow want to detect that so we can rebuild our commit index. That seems to suggest some kind of weird interaction between the commit backend and the index backend that doesn't currently exist (the index backend is agnostic to the kind of commit backend, it doesn't treat the Git backend specially in any way).

Reply:
> if someone tries to unshallow their repository we won't pick up those changes?

We might get panic somewhere because parents aren't indexed.

> Either way I'd also say for initial support that would be be probably fine.

+1. TODO comments will also help.

Reply:
Okay I just checked, we do in fact panic if the repo is deepened (`lib\src\default_index\composite.rs:472:49` in my testing just now)